Refactoring the Spec in-image examples

jordanmontt opened the pull request:
Done things

- Created dynamic layout example
- Moved all defaultLayout methods from class side to instance side
- Recategorised the classes inside the package Spec2-Examples
- Reordered the examples
- Added filtering list example
- Moved the form example to the right package
- Refactored the transmission example and added it to the spec demo presenter

![Enregistrement-de-lecran-2022-03](https://user-images.githubusercontent.com/33934979/157252404-433bf84c-fc8c-4f10-9fd8-d4c5b7756476.gif)

jordanmontt committed:
Refactoring the spec examples available in the Pharo image

jordanmontt committed:
Refactoring

Ducasse commented:
Oh this is great!!!

estebanlm commented on src/Spec2-Examples/SpApplicationWithToolbar.class.st:
```diff
@@ -42,6 +34,15 @@ SpApplicationWithToolbar >> addItemTo: aGroup [
+SpApplicationWithToolbar >> defaultLayout [
+
+	^ SpBoxLayout newTopToBottom
+		add: menu withConstraints: [ :constraints | constraints height: self class toolbarHeight ];
```
in fact... add:height: would be not correct here (`add: ... expand: false` would be)

jordanmontt replied:
Ok, I'll do that

estebanlm reviewed (changes requested):
thanks for this, is very needed!
can you replace the bad usages of add:withConstraints: to fix a height with add:expand: false, as a last step?

jordanmontt committed:
Deleted unnecessary accessors and removed the fixed constraints from the layout.

jordanmontt commented:
I push changes in my branch but the PR is not updating... Do you have any idea? @estebanlm

jordanmontt committed:
Formatting

jordanmontt commented:
@estebanlm I don't know what happened but I made a frivolous commit and now everything is updated. I replaced all `add:withConstraints:` and also deleted the unnecessary accessor methods 🙂

estebanlm commented:
you left an unused variable at: SpDemoCheckboxesPresenter :)

jordanmontt committed:
Deleted unused variable at: SpDemoCheckboxesPresenter

jordanmontt commented:
Deleted :)

Do you find something else that can be improved?
